src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs

```diff
@@ -58,6 +58,7 @@ private static async Task<int> Collect(CancellationToken ct, IConsole console, i
             bool cancelOnEnter = true;
             bool cancelOnCtrlC = true;
             bool printStatusOverTime = true;
+            bool processStarted = true;
```

Comment:
Rather than add this state here, I'd suggest adding it inside the ProcessLauncher class. A few reasons for that:
- Multiple tools use this ProcessLauncher and there is a decent chance they could all have the same bug. Doing the work inside ProcessLauncher potentially fixes it for all of them automatically or at least it should minimize the amount of change that needs to be repeated in each tool.
- The code inside ProcessLauncher potentially has better understanding whether the process did start.
- I'm not sure if this is something you saw in school (my school never showed me), but all else being equal we want to minimize coupling (https://en.wikipedia.org/wiki/Coupling_(computer_programming)). Coupling is roughly how complex the interaction between two software components is or how much component A needs to understand about component B in order to use it correctly. For example...

Higher coupling:
```C#
bool isStarted = true;
try
{
   launcher.Start();
   // do work
}
catch(SomeException) { isStarted = false; }
finally
{
    if(launcher.HasChildProc)
    {
        if(isStarted && !launcher.Process.HasExited)
        {
            launcher.Process.Kill();
        }
    }
}
```

Lower coupling alternate design:
```C#
try
{
    launcher.Start();
    // do work
}
finally
{
    // the launcher knows if the process was started or not so it can kill if needed
    launcher.Cleanup(); 
}
```